src/runtime/test/strategies/strategy-sequence-test.ts

```diff
@@ -111,6 +113,7 @@ describe('A Strategy Sequence', () => {
     recipe = await onlyResult(arc, MatchRecipeByVerb, recipe);
     recipe = await onlyResult(arc, ConvertConstraintsToConnections, recipe);
     recipe = await onlyResult(arc, ResolveRecipe, recipe);
+    recipe = await onlyResult(arc, ResolveRecipe, recipe);
```

Comment:
Can you explain why ResolveRecipe can be called twice now?

Reply:
I've added a comment to give an explanation. It's a bit verbose, let me know what you think.

Reply:
Is it the same issue as above? 
feels really weird to me that running ResolveRecipe the second time will return different results. Maybe we should encapsulate this issue inside ResolveRecipe strategy (or insider RecipeResolver maybe?) and just always execute it twice always, until the issue is resolved (instead of updating the callers)?
RecipeResolver is used in various places (where it ideally shouldn't, but..), for example, in the shell/, so  might be affected but this change...

Reply:
It is the same issue, but repeat invocation of ResolveRecipe won't always fix it, so I don't think it's worthwhile. Instead we should track the bug and fix it when slandles have been implemented (if the bug is still present).